filecoindot/src/lib.rs

```diff
@@ -64,6 +65,12 @@ pub mod pallet {
         type WeightInfo: WeightInfo;
         /// The timeout of the http requests of ocw in milliseconds
         type OffchainWorkerTimeout: Get<u64>;
+        type AMTNode: filecoindot_proofs::AMTNode;
+        type StateKey: Eq;
+        type StateVal: Eq;
+        type StateHash: filecoindot_proofs::HashedBits;
+        type HAMTNode: filecoindot_proofs::HAMTNode<Self::StateKey, Self::StateVal, Self::StateHash>;
+        type Verify: filecoindot_proofs::Verify;
```

Comment:
are those necessary, can't we just call into the crate directly, or are there some generic restrictions in `proofcrate::verify`?

Reply:
yeah, I changed this already